frontend/packages/dev-console/src/components/topology2/layoutFactory.ts

```diff
@@ -0,0 +1,16 @@
+import { Graph, Layout, LayoutFactory, ForceLayout } from '@console/topology';
+
+const layoutFactory: LayoutFactory = (type: string, graph: Graph): Layout | undefined => {
+  switch (type) {
+    case 'Force':
+      return new ForceLayout(graph, {
+        linkDistance: 80,
+        collideDistance: 30,
+        chargeStrength: -50,
+      });
+    default:
+      return undefined;
```

Comment:
Does this not just insert an empty item into your layoutFactories?

```
  registerLayoutFactory(factory: LayoutFactory) {
    this.layoutFactories.unshift(factory);
  }
```

Reply:
No because the factory is a function. A factory may not support a type and therefore return an undefined layout.